example/src/screens/CreateAPIClientScreen.tsx

```diff
@@ -0,0 +1,295 @@
+import {
+    Alert,
+    Button,
+    Platform,
+    PlatformColor,
+    SafeAreaView,
+    ScrollView,
+    StyleSheet,
+    Text,
+    TextInput,
+    View,
+} from 'react-native';
+import CheckBox from '@react-native-community/checkbox';
+import DeviceInfo from 'react-native-device-info';
+import {getOrCreateAPIClient} from '@mattermost/react-native-network-client';
+
+const styles = StyleSheet.create({
+    container: {
+        flex: 1,
+    },
+    inputContainer: {
+        flexDirection: 'row',
+        justifyContent: 'space-between',
+        alignItems: 'center',
+        padding: 10,
+    },
+    textInputContainer: {
+        flex: 5,
+    },
+    numericInputContainer: {
+        flex: 0.5,
+    },
+    label: {
+        padding: 5,
+        flex: 1,
+    },
+    input: {
+        margin: 15,
+        height: 40,
+        borderWidth: 1,
+        padding: 5,
+        ...Platform.select({
+            ios: { borderColor: PlatformColor('link') },
+            android: {
+                borderColor: PlatformColor('?attr/colorControlNormal')
+            },
+        }),
+     },
+     headersLabelContainer: {
+        flexDirection: 'row',
+        alignItems: 'center',
+     },
+     headersContainer: {
+        flex: 8,
+     },
+     header: {
+        flexDirection: 'row',
+        justifyContent: 'space-evenly',
+     },
+});
+
+const ClientHeader = ({index, header, updateHeader}) => {
+    const [key, setKey] = useState(header.key);
+    const [value, setValue] = useState(header.value);
+
+    const doUpdateHeader = () => {
+        updateHeader(index, {key, value});
+    }
+
+    return (
+        <>
+            <TextInput
+                value={key}
+                onChangeText={setKey}
+                placeholder='key'
+                autoCapitalize='none'
+                onBlur={doUpdateHeader}
+                style={[styles.input, {flex: 1}]}
+            />
+            <TextInput
+                value={value}
+                onChangeText={setValue}
+                placeholder='value'
+                autoCapitalize='none'
+                onBlur={doUpdateHeader}
+                style={[styles.input, {flex: 1}]}
+            />
+        </>
+    );
+
+}
+
+export default function CreateAPIClientScreen({navigation}) {
+    const [name, setName] = useState('HTTP Google Redirect Test');
+    const [baseUrl, setbaseUrl] = useState('http://google.com');
+    const [clientHeaders, setClientHeaders] = useState([]);
+    const [followRedirects, setFollowRedirects] = useState(true);
+    const [allowsCellularAccess, setAllowsCellularAccess] = useState(true);
+    const [waitsForConnectivity, setWaitsForConnectivity] = useState(false);
+    const [timeoutIntervalForRequest, setTimeoutIntervalForRequest] = useState('60');
+    const [timeoutIntervalForResource, setTimeoutIntervalForResource] = useState('60');
+    const [httpMaximumConnectionsPerHost, setHttpMaximumConnectionsPerHost] = useState('10');
```

Comment:
Would it be worth combining all of these into a single `options` or `params` object?